contracts/smart-wallet/managed/TWManagedAccountFactory.sol

```diff
@@ -0,0 +1,87 @@
+// SPDX-License-Identifier: GPL-3.0
+pragma solidity ^0.8.12;
+
+// Utils
+
+import "../utils/BaseRouter.sol";
+import "../../extension/Multicall.sol";
+import "@openzeppelin/contracts/proxy/Clones.sol";
+import "../../dynamic-contracts/extension/PermissionsEnumerable.sol";
+
+// Interface
+import "../interfaces/ITWAccountFactory.sol";
+
+// Smart wallet implementation
+import "./TWManagedAccount.sol";
+
+//   $$\     $$\       $$\                 $$\                         $$\
+//   $$ |    $$ |      \__|                $$ |                        $$ |
+// $$$$$$\   $$$$$$$\  $$\  $$$$$$\   $$$$$$$ |$$\  $$\  $$\  $$$$$$\  $$$$$$$\
+// \_$$  _|  $$  __$$\ $$ |$$  __$$\ $$  __$$ |$$ | $$ | $$ |$$  __$$\ $$  __$$\
+//   $$ |    $$ |  $$ |$$ |$$ |  \__|$$ /  $$ |$$ | $$ | $$ |$$$$$$$$ |$$ |  $$ |
+//   $$ |$$\ $$ |  $$ |$$ |$$ |      $$ |  $$ |$$ | $$ | $$ |$$   ____|$$ |  $$ |
+//   \$$$$  |$$ |  $$ |$$ |$$ |      \$$$$$$$ |\$$$$$\$$$$  |\$$$$$$$\ $$$$$$$  |
+//    \____/ \__|  \__|\__|\__|       \_______| \_____\____/  \_______|\_______/
+
+contract TWManagedAccountFactory is ITWAccountFactory, Multicall, PermissionsEnumerable, BaseRouter {
+    /*///////////////////////////////////////////////////////////////
+                                State
+    //////////////////////////////////////////////////////////////*/
+
+    TWManagedAccount private immutable _accountImplementation;
+
+    /*///////////////////////////////////////////////////////////////
+                            Constructor
+    //////////////////////////////////////////////////////////////*/
+
+    constructor(IEntryPoint _entrypoint) {
+        _setupRole(DEFAULT_ADMIN_ROLE, msg.sender);
+        _accountImplementation = new TWManagedAccount(_entrypoint);
+    }
+
+    /*///////////////////////////////////////////////////////////////
+                        External functions
+    //////////////////////////////////////////////////////////////*/
+
+    /// @notice Deploys a new Account with the given admin and accountId used as salt.
+    function createAccount(address _admin, string memory _accountId) external returns (address) {
+        address impl = address(_accountImplementation);
+        bytes32 salt = keccak256(abi.encode(_accountId));
+        address account = Clones.predictDeterministicAddress(impl, salt);
+
+        if (account.code.length > 0) {
+            return account;
+        }
+
+        account = Clones.cloneDeterministic(impl, salt);
+
+        TWManagedAccount(payable(account)).initialize(_admin);
```

Comment:
would it make sense to add the factory as admin of the accounts too? wondering about the "recovery" use case.